networkx/algorithms/centrality/trophic.py

```diff
@@ -40,6 +40,29 @@ def trophic_levels(G, weight="weight"):
     ----------
     .. [1] Stephen Levine (1980) J. theor. Biol. 83, 195-207
     """
+
+    def bfs_reachable_from_basal(basal_nodes, graph):
+        reachable_nodes = {
+            node
+            for layer in nx.bfs_layers(graph, sources=basal_nodes)
+            for node in layer
+        }
```

Comment:
I'd vote to use this directly on L59 and do away with the extra function definition - eliminating indirection makes the code much easier to follow IMO!